lib/src/main/java/edu/stanford/futuredata/macrobase/operator/Operator.java

```diff
@@ -1,5 +1,13 @@
 package edu.stanford.futuredata.macrobase.operator;
 
+/**
+ * Generic interface for operations. Should not modify input.
```

Comment:
Understood; wasn't thinking about column storage.

However, I _was_ thinking of in-place transforms to metrics; won't that be expensive? (Probably overthinking here.)